src/ck3/data/characters.rs

```diff
@@ -435,50 +413,54 @@ impl Character {
                     vd.req_field("death_reason");
                     vd.field_item("death_reason", Item::DeathReason);
                     vd.field_item("killer", Item::Character);
-                    Some((Death, key))
+                    return Some((Death, key.clone()));
                 }
                 "effect" => {
                     validate_effect(block, data, sc, Tooltipped::No);
-                    None
-                }
-                _ => {
-                    let msg = format!("unknown key `{key}` in definition");
-                    err(ErrorKey::Validation).msg(msg).loc(key).push();
-                    None
+                    return None;
                 }
+                _ => (),
             },
         };
 
-        result.map(|(etype, token)| (etype, token.clone()))
+        // unknown effect field
+        validate_effect_field(
+            &crate::lowercase::Lowercase::empty(),
+            key,
+            crate::block::Comparator::Equals(crate::block::Eq::Single),
```

Comment:
Importing just `Eq::Single` is done in some other files too, you can see how it's usually done